### PR TITLE
docs: `op-succinct` End to End tutorial

### PR DESCRIPTION
**Pull request by zobront**
Add end to end tutorial for upgrading an existing OP Stack chain to use validity proofs with `op-succinct`.

### CODE REVIEW

File: contracts/README.md [hunk 1]
```diff
@@ -38,5 +38,5 @@ The `ZKL2OutputOracle.sol` contract requires a verification key to be able to ve
 This key can be generated by running the following command from the root of the repo:
 
 ```shell
-just generate-agg-vkey
+cargo run --bin vkey
```
Comment (ratankaliani) [inline contracts/README.md:41]: `cargo run --bin vkey --release`

Comment (zobront) [reply contracts/README.md]: Fixed.

File: op-succinct-proposer/TUTORIAL.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+4. `op-proposer`: Posts an output root to L1 at regular intervals, which captures the L2 state so withdrawals can be processed.
+
+## Step-by-Step Guide
+
+### Step 1: Setup
+
+1. Clone the `op-succinct` repo:
+   ```
+   git clone https://github.com/succinctlabs/op-succinct.git
+   ```
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:33]: Might want to create a `tutorial` branch and have them check that out, so the tutorial doesn't always have to stay up to date with future changes that get pushed to `main`.

File: op-succinct-proposer/TUTORIAL.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    <details>
+    <summary>Field Info</summary>
+
+    - `startingBlockNumber`: The L1 block number at which the rollup starts. Default should be 0.
+    - `l2RollupNode`: The URL of the L2 rollup node. (After the tutorial, this is `http://localhost:8545`)
+    - `submissionInterval`: The number of L2 blocks between each L1 output submission.
+    - `l2BlockTime`: The time in seconds between each L2 block.
+    - `proposer`: The Ethereum address of the proposer account. Set to `address(0)` for anyone to submit proofs.
+    - `challenger`: The Ethereum address of the challenger account. Set to `address(0)` for anyone to undo proofs.
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:51]: I don't think anyone would want the ability for anyone to undo proofs lol! If it's set to `address(0)`,  nobody can undo proofs (ie the safeguards to undo malicious proposals are turned off).

File: op-succinct-proposer/TUTORIAL.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    - `finalizationPeriod`: The time period (in seconds) after which a proposed output becomes finalized.
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:52]: Specifically, this is how long after an output root is finalized that you can do a withdrawal against it.

File: op-succinct-proposer/TUTORIAL.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    - `chainId`: The chain ID of the L2 network.
+    - `owner`: The Ethereum address of the contract owner.
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:54]: Owner of the ZKL2OutputOracle, who can update vkey and verifier gateway.

File: op-succinct-proposer/TUTORIAL.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+   forge script script/ZKUpgrader.s.sol:ZKUpgrader --rpc-url <L1 RPC> --private-key <ADMIN PK> --verify --verifier etherscan --etherscan-api-key <ETHERSCAN API> --broadcast --slow --vvvv
+   ```
+   Alternatively, use:
+   ```
+   just upgrade-l2oo <L1_RPC> <ADMIN_PK> <ETHERSCAN_API_KEY>
+   ```
+
+To run your OP Stack chain with ZK proofs you need to replace your chain's `L2OutputOracle` with `ZKL2OutputOracle`. 
+
+The existing `L2OutputOracle` allows a permissioned `proposer` role to submit output roots. The permissioned `challenger` role can dispute these output roots. Theree are no checks on the validity of these claims.
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:79]: Theree

File: op-succinct-proposer/TUTORIAL.md (hunk 2 above)
Comment (ratankaliani) [reply op-succinct-proposer/TUTORIAL.md:33]: The tutorial should always work on `main` IMO, it's better practice that way. We can add a CI test that confirms that parts of the tutorial work correctly.

File: op-succinct-proposer/TUTORIAL.md (hunk 3 above)
Comment (ratankaliani) [reply op-succinct-proposer/TUTORIAL.md:51]: Haha good catch

File: op-succinct-proposer/TUTORIAL.md (hunk 2 above)
Comment (ratankaliani) [reply op-succinct-proposer/TUTORIAL.md:33]: Feels a bit weird to periodically upstream into `tutorial`, easier to just have `main` be stable after a certain point for the tutorial.

File: op-succinct-proposer/TUTORIAL.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+   cp .env.example .env
+   cp .env.server.example .env.server
+   ```
+
+3. Fill in the `contracts/zkconfig.json` file with your chain's specific details.
+
+    <details>
+    <summary>Field Info</summary>
+
+    - `startingBlockNumber`: The L1 block number at which the rollup starts. Default should be 0.
```
Comment (zobront) [inline op-succinct-proposer/TUTORIAL.md:46]: This should be L2 block number. I assume it was my mistake, just noting here to fix before merging.

**Pull request merged**